force-app/main/default/classes/RD2_ScheduleService.cls

```diff
@@ -187,6 +203,7 @@ public without sharing class RD2_ScheduleService {
         List<RecurringDonationSchedule__c> newSchedules = buildNewSchedules(rd);
         for (RecurringDonationSchedule__c newSchedule : newSchedules) {
             newSchedule.StartDate__c = newStartDate;
+            newSchedule.NextPaymentAmount__c = nextPaymentAmount;
```

Comment:
What's the reason for putting nextPaymentAmount on more than one schedule record?

Reply:
No reason, in my testing this was always a single schedule, but I didn't account for 1st and 15th. Are there other scenarios where there will be more than 1 new schedule created?

Reply:
I believe that right now that's the only time we create more than one.

There can be multiple active schedules: 2 current if it's a 1st and 15th schedule + 2 future if the future schedule is a 1st and 15th + pause schedule. Currently, five active would be the cap I think, but there's no structural reason we would be prohibited from having more future schedules or more current schedules. Unlikely we'll do either of those things, but you never know.

Reply:
I'm refactoring this a bit, but I'm still going to set NextPaymentAmount on all new Schedules. It shouldn't hurt anything if we clear them all after that Payment is paid.